crates/mock-download-server/src/lib.rs

```diff
@@ -44,10 +48,11 @@ pub struct Data {
 
 pub struct Builder {
     data: Data,
+    routes: fn() -> Router<Arc<Mutex<Data>>>,
 }
 
 impl Builder {
-    pub fn new() -> Builder {
+    pub fn new(routes: fn() -> Router<Arc<Mutex<Data>>>) -> Builder {
```

Comment:
Huh I'm actually surprised this works, RUst has really gotten better over the years.